autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/load/YarnLoadEvaluator.java

```diff
@@ -118,74 +125,105 @@ protected void execute() {
         Optional<Integer> mandatoryDownScaleCount = Optional.of(configMaxNodeCount)
                 .filter(mandatoryDownscale -> mandatoryDownscale < 0).map(downscale -> -1 * downscale);
 
-        YarnScalingServiceV1Response yarnResponse = yarnMetricsClient
-                .getYarnMetricsForCluster(cluster, stackV4Response, policyHostGroup, pollingUserCrn, mandatoryDownScaleCount);
+        Optional<Integer> mandatoryStopStartUpscaleCount = isStopStartEnabled ? Optional.of(stopStartAdjustment)
+                .filter(mandatoryStopStartUpscale -> mandatoryStopStartUpscale > 0) : Optional.empty();
+        Optional<Integer> mandatoryStopStartDownscaleCount = isStopStartEnabled ? Optional.of(stopStartAdjustment)
+                .filter(mandatoryStopStartDownscale -> mandatoryStopStartDownscale < 0).map(Math::abs) : Optional.empty();
+
+        YarnScalingServiceV1Response yarnResponse =
+                yarnMetricsClient.getYarnMetricsForCluster(cluster, stackV4Response, policyHostGroup, pollingUserCrn, mandatoryDownScaleCount);
 
         if (cluster.getUpdateFailedDetails() != null && pollingUserCrn.equals(cluster.getMachineUserCrn())) {
             // Successful YARN API call
             clusterService.setUpdateFailedDetails(cluster.getId(), null);
         }
 
-        int yarnRecommendedScaleUpCount = yarnResponseUtils.getYarnRecommendedScaleUpCount(yarnResponse, policyHostGroup,
-                maxAllowedUpScale, mandatoryUpScaleCount, loadAlertConfiguration.getMaxScaleUpStepSize());
-        List<String> yarnRecommendedDecommissionHosts = yarnResponseUtils.
-                getYarnRecommendedDecommissionHostsForHostGroup(cluster.getStackCrn(), yarnResponse,
-                        hostFqdnsToInstanceId, maxAllowedDownScale, mandatoryDownScaleCount, loadAlertConfiguration.getMaxScaleDownStepSize());
+        int yarnRecommendedScaleUpCount = yarnResponseUtils.getYarnRecommendedScaleUpCount(yarnResponse, policyHostGroup, maxAllowedUpScale,
+                mandatoryUpScaleCount, loadAlertConfiguration.getMaxScaleUpStepSize(), mandatoryStopStartUpscaleCount);
+        List<String> yarnRecommendedDecommissionHosts =
+                yarnResponseUtils.getYarnRecommendedDecommissionHostsForHostGroup(yarnResponse, hostFqdnsToInstanceId, stoppedHostInstanceIds,
+                        maxAllowedDownScale, mandatoryDownScaleCount, mandatoryStopStartDownscaleCount, loadAlertConfiguration.getMaxScaleDownStepSize());
 
-        LOGGER.info("yarnRecommendedScaleUpCount={}, yarnRecommendedDecommssion={}", yarnRecommendedScaleUpCount, yarnRecommendedDecommissionHosts);
+        LOGGER.info("yarnRecommendedScaleUpCount={}, yarnRecommendedDecommission={}", yarnRecommendedScaleUpCount, yarnRecommendedDecommissionHosts);
+
+        ScalingAdjustmentType adjustmentType = !isStopStartEnabled || mandatoryStopStartUpscaleCount.isPresent() || mandatoryStopStartDownscaleCount.isPresent()
```

Comment:
Is this handling the scenario where we need to upscale because there's not enough nodes, but the cluster has stopped instances - so those need to be started as a first step.

Reply:
Updated the patch to perform this prerequisite step before adding new instances to the cluster